usaspending_api/search/v2/views/spending_by_geography.py

```diff
@@ -66,6 +66,26 @@ class SpendingByGeographyVisualizationViewSet(APIView):
 
     @cache_response()
     def post(self, request: Request) -> Response:
+        # First determine if we are using Subaward or Prime Awards / Transactions as this
+        # will impact some of the downstream filters in the JSON request
+        spending_type_models = [
+            {"name": "subawards", "key": "subawards", "type": "boolean", "default": False},
+            {
+                "name": "spending_level",
+                "key": "spending_level",
+                "type": "enum",
+                "enum_values": [level.value for level in SpendingLevel],
+                "optional": True,
+                "default": "transactions",
+            },
+        ]
+        spending_level_tiny_shield = TinyShield(spending_type_models)
+        spending_level_data = spending_level_tiny_shield.block(request.data)
+        self.spending_level = SpendingLevel(
+            "subawards" if spending_level_data["subawards"] else spending_level_data["spending_level"]
+        )
```

Comment:
I like this way of validating either the `subawards` or `spending_level` filter for subawards.

Reply:
This was noticed during Evan's work on the spending_by_award endpoint. Ideally, I would like to break this out independent of the endpoint, but I am going to repurpose the ES Feature Flag ticket to instead cleanup our handling of Tiny Shield across the spending endpoints.

That is also why I am ignoring the code complexity warning in codeclimate right now in favor of making all of this more DRY.